### PR TITLE
Explicitly order courses instead of alphabetizing

### PR DESCRIPTION
**Pull request by balderdash**
Ignoring the moratorium on staging merges for this PR because it's wanted for facilitator summit.

Alphabetical is inadequate, e.g. 'Post-AP' shouldn't be at the beginning of CSP:

Post-AP: Databases in Apps
Unit 1: Digital Information
Unit 2: The Internet
etc.

This uses the order of the courses in the `ScriptConstant::CATEGORIES` map to sort them in the dropdown instead:

image

Also fix a bug that wasn't showing nice names for hidden courses, and sort categories alphabetically (I'm pretty sure categories won't need custom ordering as well).

### CODE REVIEW

File: lib/cdo/script_constants.rb [hunk 1]
```diff
@@ -102,6 +101,10 @@ def self.categories(script)
         map {|category, _| category.to_s}
   end
 
+  def self.position_in_category(script, category)
+    CATEGORIES[category.to_sym] ? CATEGORIES[category.to_sym].find_index(script) : -1
```
Comment (Hamms) [inline lib/cdo/script_constants.rb]: Why `-1` rather than `nil`?

File: pegasus/test/test_script_constants.rb [hunk 2]
```diff
@@ -44,4 +44,13 @@ def test_category
...
     assert_equal ['csp'], ScriptConstants.categories(ScriptConstants::CSP_UNIT1_NAME)
   end
+
+  def test_category_index
+    assert_equal 0, ScriptConstants.position_in_category(ScriptConstants::CSP_UNIT1_NAME, :csp)
+    assert_equal 1, ScriptConstants.position_in_category(ScriptConstants::CSP_UNIT2_NAME, :csp)
+    assert_equal 2, ScriptConstants.position_in_category(ScriptConstants::CSP_UNIT3_NAME, :csp)
+    assert_equal 3, ScriptConstants.position_in_category(ScriptConstants::CSP_UNIT4_NAME, :csp)
+    assert_equal 4, ScriptConstants.position_in_category(ScriptConstants::CSP_UNIT5_NAME, :csp)
+    assert_equal 5, ScriptConstants.position_in_category(ScriptConstants::CSP_UNIT6_NAME, :csp)
```
Comment (Hamms) [inline pegasus/test/test_script_constants.rb:54]: can we throw in a test for the "script not in category" case?

**Comment by Hamms** [comment]
LGTM after comments

File: pegasus/test/test_script_constants.rb (hunk 2 above)
Comment (balderdash) [inline pegasus/test/test_script_constants.rb:54]: Yep.

File: lib/cdo/script_constants.rb (hunk 1 above)
Comment (balderdash) [inline lib/cdo/script_constants.rb]: Because I didn't realize that was the convention in ruby. Switched to nil.

**Pull request merged**